### PR TITLE
Оптимизируй это (Часть 2)

### PR DESCRIPTION
**Pull request by olegsorokin**
---
:mortar_board: [Оптимизируй это (Часть 2)](https://up.htmlacademy.ru/react/9/user/30818/tasks/14)

### CODE REVIEW

File: project/src/components/catalog/catalog.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+type Props = {
+  films: Film[],
+};
+
+function Catalog({ films }: Props) {
+  const [activeGenre, setActiveGenre] = useState<GenreEnum>(GenreEnum.ALL_GENRES);
+
+  const onGenreChange = (genreName: GenreEnum): void => {
```
Comment (dianatsevan) [inline project/src/components/catalog/catalog.tsx:23]: do we need to optimize it? if yes - why and how to? if no - why no?

File: project/src/pages/main/main.tsx [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
 
-  const onGenreChange = (genreName: GenreEnum): void => {
-    setActiveGenre(genreName);
+  const handleMyPlaylistButtonClick = async () => {
+    await dispatch(addToFavorite({
+      filmId: Number(promo.data?.id),
+      status: promo.data?.isFavorite ? 0 : 1,
+    }));
+    await dispatch(fetchPromo());
   };
```
Comment (dianatsevan) [inline project/src/pages/main/main.tsx:26]: Copypaste as I see. Consider using custom hook or standalone component to encapsulate the logic

File: project/src/utils/common.ts [hunk 3]
```diff
@@ -36,3 +38,42 @@ export function loadData<T>(state: LoadingState<T>, type: PromiseState): Loading
...
 }
+
+enum RatingLevel {
+  UNKNOWN = 'Unknown',
+  BAD = 'Bad',
+  NORMAL = 'Normal',
+  GOOD = 'Good',
+  VERY_GOOD = 'Very good',
+  AWESOME = 'Awesome',
+}
```
Comment (dianatsevan) [inline project/src/utils/common.ts:49]: place it in constants, it is not a util func

File: project/src/components/catalog/catalog.tsx (hunk 1 above)
Comment (olegsorokin) [reply project/src/components/catalog/catalog.tsx:23]: Yes! This optimization works!

**Pull request merged**